services/src/model_training/ml_tasks.rs

```diff
@@ -0,0 +1,177 @@
+pub struct MLTrainRequest {
+    pub params: XgboostTrainingParams,
+    pub input_workflows: Vec<Workflow>,
+    pub label_workflow: Vec<Workflow>,
+    pub query: MachineLearningQueryRectangle,
+}
+
+/// parameter for the dataset from workflow handler (body)
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+#[schema(example = json!({"name": "foo", "description": null, "query": {"spatialBounds": {"upperLeftCoordinate": {"x": -10.0, "y": 80.0}, "lowerRightCoordinate": {"x": 50.0, "y": 20.0}}, "timeInterval": {"start": 1_388_534_400_000_i64, "end": 1_388_534_401_000_i64}, "spatialResolution": {"x": 0.1, "y": 0.1}}}))]
+pub struct MachineLearningModelFromWorkflow {
+    pub name: String,
+    pub description: Option<String>,
+    pub query: MachineLearningQueryRectangle,
+
+    // TODO: is that cog stuff relevant in this case?
+    #[schema(default = default_as_cog)]
+    #[serde(default = "default_as_cog")]
+    pub as_cog: bool,
+}
+
+/// response of the machine learning model from workflow handler
+#[cfg(feature = "xgboost")]
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+pub struct MachineLearningModelFromWorkflowResult {
+    pub store_path: PathBuf,
+    pub model: Value,
+}
+
+#[cfg(feature = "xgboost")]
+impl TaskStatusInfo for MachineLearningModelFromWorkflowResult {}
+
+#[cfg(feature = "xgboost")]
+pub struct MachineLearningModelFromWorkflowTask<C: Context> {
+    pub input_workflows: Vec<Workflow>,
+    pub label_workflow: Vec<Workflow>,
+    pub session: C::Session,
+    pub ctx: Arc<C>,
+    pub info: MLTrainRequest,
+    pub query: MachineLearningQueryRectangle,
+    pub upload: UploadId, // TODO: remove?
+    pub store_path: PathBuf,
+}
+
+#[cfg(feature = "xgboost")]
+impl<C: Context> MachineLearningModelFromWorkflowTask<C> {
+    async fn process(&self) -> error::Result<MachineLearningModelFromWorkflowResult> {
+        let train_cfg = self.info.params.training_config.clone();
+        let store_path = self.store_path.clone();
+        let feature_names = self.info.params.feature_names.clone();
+
+        // put the labels workflow at the end of the workflow vec
+        let mut inputs = self.info.input_workflows.clone();
+        let labels = self.info.label_workflow.clone();
+        inputs.extend(labels);
+
+        let input_operators = get_operators_from_workflows(inputs)?;
+
+        let mut exe_ctx = self.ctx.execution_context(self.session.clone())?;
+
+        let typed_query_processors = get_query_processors(input_operators, &exe_ctx).await?;
+
+        let query = self.query;
+        let query_ctx = self.ctx.query_context(self.session.clone())?;
+
+        let mut accumulated_data =
+            accumulate_raster_data(feature_names, typed_query_processors, query, &query_ctx)
+                .await?;
```

Comment:
maybe you should check for Errors already here. So turn the `Vec<Result>` into a `Result<Vec>` and unpack it